bento/crates/workflow/src/tasks/keccak.rs

```diff
@@ -51,28 +51,58 @@ pub async fn keccak(
 
     tracing::info!("Keccak proving {}", request.claim_digest);
 
+    // Main computational work
     let keccak_receipt = agent
         .prover
         .as_ref()
         .context("Missing prover from keccak prove task")?
         .prove_keccak(&keccak_req)
         .context("Failed to prove_keccak")?;
 
+    // Clone data needed for background task
+    let pool = agent.redis_pool.clone();
     let job_prefix = format!("job:{job_id}");
-    let receipts_key = format!("{job_prefix}:{KECCAK_RECEIPT_PATH}:{task_id}");
-    let keccak_receipt_bytes =
-        serialize_obj(&keccak_receipt).context("Failed to serialize keccak receipt")?;
+    let receipts_key = format!("{job_prefix}:{RECEIPT_PATH}:{task_id}");
+    let ttl = agent.args.redis_ttl;
+    let claim_digest = request.claim_digest;
 
-    redis::set_key_with_expiry(
-        &mut conn,
-        &receipts_key,
-        keccak_receipt_bytes,
-        Some(agent.args.redis_ttl),
-    )
-    .await
-    .context("Failed to write keccak receipt to redis")?;
+    // Spawn background task for serialization and storage
+    tokio::spawn(async move {
```

Comment:
i don't think we benefit from using this here, imho spawn makes sense when we have multiple tasks